### PR TITLE
feat: button confirmations

### PR DESCRIPTION
**Pull request by EvolutionX-10**

### CODE REVIEW

File: TypeScript/buttonConfirmation.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+import { CommandType, EventPlugin, PluginType } from "@sern/handler";
+import {
+	ActionRowBuilder,
+	ButtonBuilder,
+	ButtonStyle,
+	ComponentType,
+} from "discord.js";
+
+export function confirmation(
+	options?: Partial<ConfirmationOptions>
```
Comment (jacoobes) [inline TypeScript/buttonConfirmation.ts:31]: add default options here to avoid short circuiting everywhere

Comment (jacoobes) [reply TypeScript/buttonConfirmation.ts:31]:
```ts
options = {
   content : ''
} 
```

Comment (jacoobes) [reply TypeScript/buttonConfirmation.ts:31]: etc etc

**Review by jacoobes — CHANGES_REQUESTED** [review]

File: TypeScript/buttonConfirmation.ts [hunk 2]
```diff
@@ -87,7 +94,7 @@ export function confirmation(
 
 				collector.on("ignore", async (i) => {
 					await i.reply({
-						content: options?.wrongUserResponse ?? "Not for you!",
+						content: options?.wrongUserResponse,
```
Comment (jacoobes) [inline TypeScript/buttonConfirmation.ts:97]: could remove the ? in ?. but its fine

**Review by jacoobes — APPROVED** [review]

File: TypeScript/buttonConfirmation.ts (hunk 2 above)
Comment (EvolutionX-10) [reply TypeScript/buttonConfirmation.ts:97]: image

**Pull request merged**